### PR TITLE
feat: migrate use chatbot mutation

### PR DESCRIPTION
**Pull request by ReidyT**
issue #70

### CODE REVIEW

File: src/components/common/chat/ChatBox.tsx [hunk 1]
```diff
@@ -13,19 +13,20 @@ import {
...
+import { CHAT_BOT_ERROR_MESSAGE } from '../../../config/messages';
+import { mutations } from '../../../config/queryClient';
 import { CHATBOT_MODE_CY, messagesDataCy } from '../../../config/selectors';
 import {
   DEFAULT_BORDER_RADIUS,
   LIGHT_GRAY,
   LIGHT_VIOLET,
 } from '../../../config/stylingConstants';
+import { ThreadMessage } from '../../../interfaces/chatbot';
+import { buildPrompt } from '../../../utils/chatbot';
```
Comment (spaenleh) [inline src/components/common/chat/ChatBox.tsx:29]: This looks like it is used in all the apps that use the chatbot, should it be exported as a util function from somewhere common ?

File: src/utils/chatbot.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { ChatBotMessage, ChatbotRole } from '@graasp/sdk';
+
+import { APP_DATA_TYPES } from '../config/appDataTypes';
+import { ThreadMessage } from '../interfaces/chatbot';
+
+// TODO: maybe it is possible to move to SDK and to reuse it in code-capsule
```
Comment (spaenleh) [inline src/utils/chatbot.ts:6]: Yes that would be nice ! (Can be done in another PR)

Comment (ReidyT) [reply src/utils/chatbot.ts:6]: I opened an issue in the apps-query-client and I will migrate in another PR.

File: .github/workflows/deploy-dev.yml [hunk 3]
```diff
@@ -41,6 +40,6 @@ jobs:
           version: 'latest'
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_DEV }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_DEV }}
-          aws-region: ${{ secrets.APPS_AWS_REGION }}
-          aws-s3-bucket-name: ${{ secrets.AWS_S3_BUCKET_NAME_APPS_DEV }}
+          aws-region: ${{ vars.APPS_AWS_REGION }}
```
Comment (spaenleh) [inline .github/workflows/deploy-dev.yml:43]:
```suggestion
          aws-region: ${{ vars.APPS_AWS_REGION_DEV }}
```

File: .github/workflows/deploy-prod.yml [hunk 4]
```diff
@@ -40,6 +39,6 @@ jobs:
           version: 'latest'
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_PROD }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_PROD }}
-          aws-region: ${{ secrets.APPS_AWS_REGION }}
-          aws-s3-bucket-name: ${{ secrets.AWS_S3_BUCKET_NAME_APPS_PROD }}
+          aws-region: ${{ vars.APPS_AWS_REGION }}
```
Comment (spaenleh) [inline .github/workflows/deploy-prod.yml:42]:
```suggestion
          aws-region: ${{ vars.APPS_AWS_REGION_PROD }}
```

File: .github/workflows/deploy-stage.yml [hunk 5]
```diff
@@ -40,6 +39,6 @@ jobs:
           version: 'latest'
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_STAGE }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_STAGE }}
-          aws-region: ${{ secrets.APPS_AWS_REGION }}
-          aws-s3-bucket-name: ${{ secrets.AWS_S3_BUCKET_NAME_APPS_STAGE }}
+          aws-region: ${{ vars.APPS_AWS_REGION }}
```
Comment (spaenleh) [inline .github/workflows/deploy-stage.yml:42]:
```suggestion
          aws-region: ${{ vars.APPS_AWS_REGION_STAGE }}
```

**Review by spaenleh — CHANGES_REQUESTED** [review]
Almost there ! 💪

**Review by spaenleh — APPROVED** [review]

**Pull request merged**